File: frontend/src/api/client/apiClient.ts
import axios from "axios";
import { API_CONFIG } from "./apiConfig";

export const apiClient = axios.create({
    baseURL: API_CONFIG.baseURL,
    timeout: API_CONFIG.timeout,
    headers: {
        "Content-Type": "application/json",
        Accept: "text/plain",
    }
});

apiClient.interceptors.request.use(
    (config) => {
        const accessToken = localStorage.getItem("accessToken");

        if (accessToken) {
            config.headers.Authorization = `Bearer ${accessToken}`;
        }

        return config;
    },
    (error) => Promise.reject(error)
);

File: mobile/lib/services/api_client.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ApiClient {
  // backend url
  // static const String baseUrl = 'https://10.0.2.2:7131/api/v1';
  static const String baseUrl = 'https://localhost:7131/api/v1';

  Future<http.Response> post(
    String endpoint, {
    Map<String, dynamic>? body,
    String? accessToken,
  }) async {
    final headers = {
      'Content-Type': 'application/json',
      'Accept': 'text/plain',
    };

    if (accessToken != null && accessToken.isNotEmpty) {
      headers['Authorization'] = 'Bearer $accessToken';
    }

    final response = await http.post(
      Uri.parse('$baseUrl$endpoint'),
      headers: headers,
      body: body != null ? jsonEncode(body) : null,
    );

    return response;
  }
}